### PR TITLE
EO Executive Order - Build Infrastructure and Operations

### PR DESCRIPTION
**Pull request by moljac**
*   Context: Section 4.e.i.D of 

    https://eng.ms/docs/initiatives/executive-order/executive-order-requirements/executiveorderoncybersecurity/buildinfraops#build-system-security-requirements

*   Context: 

    https://microsoft.sharepoint.com/❌/r/teams/1ES2/_layouts/15/doc2.aspx?sourcedoc=%7B97E1B064-A768-4B0F-BEB5-4BF21577F3D6%7D&file=9d770e15-6208-4284-b347-b2762803623b.xlsx

Adds support to `build.cake` Cake script to generate a `buildtoolsinventory.csv` file
which contains information about various build tools repository depends on.

Items currently included are 

*   dotnet, 

*   Android SDK and NDK packages,

*   Open JDK, 

*   Mono MDK, 

*   various tools (`brew`)

### CODE REVIEW

File: utilities.cake [hunk 1]
```diff
@@ -1272,6 +1273,272 @@ Rejected / Duplicates: %PackagesRejectedDuplicates.Count%
...
+                if (line.Contains("javac"))
+                {
+                    tool = line.Replace("javac ", "");
+                    version = tool;
+                    sb.AppendLine($"javac, {version}");
+                }
+            }
+
+			System.IO.File.WriteAllText("./output/buildtoolsinventory.csv", sb.ToString());
+			System.IO.File.WriteAllText("./docs/buildtoolsinventory.csv", sb.ToString());
```
Comment (pjcollins) [inline utilities.cake:1533]: Could we upload this as a pipeline artifact so that the latest versions produced in CI can be downloaded?

Comment (moljac) [reply utilities.cake:1548]: in `./output`

```
./output/buildtoolsinventory.csv
./output/buildtoolsinventory.md
```

**Review by pjcollins — APPROVED** [review]
I think we probably don't need to commit the `docs/buildtoolsinventory.csv` and `docs/buildtoolsinventory.md` files to the repo if the generated versions are uploaded as build artifacts, but otherwise this seems good to me.

**Comment by moljac** [comment]
> I think we probably don't need to commit the `docs/buildtoolsinventory.csv` and `docs/buildtoolsinventory.md` files to the repo if the generated versions are uploaded as build artifacts, but otherwise this seems good to me.

How can I commit changed file[a] from CI?
Manually download and then commit?

Currently I commited local build[s] inventory data.

**Comment by moljac** [comment]
I get often questions about in-depth about building the repo, so having these files and adding some more details around that might be good. Yes I know I would need to update them somehow (for now manually).

**Pull request merged**